compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -2945,6 +2945,33 @@ impl<'tcx> Ty<'tcx> {
             _ => false,
         }
     }
+
+    pub fn is_known_rigid(self) -> bool {
```

Comment:
Would be good to have a doc comment explaining what "rigid" means.

For instance, `(_, _)` would be considered rigid by this check, right?

Reply:
Yeah, I can add this.